### ISSUE TITLE
Nitrogen. Takes longer time to run commands on py3 than py2.

### ISSUE THREAD

**Issue by frogunder** [saltstack/salt#41391]
Running tests on Nitrogen I see that it is taking longer to run commands on a py3 setup than a py2 setup.

### Setup
Set up a py3 master/minion environment and a py2 master/minion environment.
On both setups I installed salt with pip/pip3.

### Steps to Reproduce Issue
Run commands and see the time.

py2
```
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m0.701s
user	0m0.470s
sys	0m0.057s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m0.696s
user	0m0.457s
sys	0m0.067s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m0.694s
user	0m0.460s
sys	0m0.063s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m0.705s
user	0m0.470s
sys	0m0.063s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m0.705s
user	0m0.470s
sys	0m0.057s
root@localhost:~#

Salt Version:
           Salt: 2017.5.0-97-gdb109a3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (default, Nov 19 2016, 06:48:10)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.5.1
            ZMQ: 4.1.6
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.9.15-x86_64-linode81
         system: Linux
        version: Ubuntu 16.04 denial
```

py3
```
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m1.270s
user	0m1.020s
sys	0m0.053s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m1.161s
user	0m0.890s
sys	0m0.073s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m1.235s
user	0m0.973s
sys	0m0.063s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m1.526s
user	0m1.173s
sys	0m0.100s
root@localhost:~# time salt ubu* test.ping
ubuntu.members.linode.com:
    True

real	0m1.372s
user	0m1.103s
sys	0m0.043s
root@localhost:~# 

Salt Version:
           Salt: 2017.5.0-97-gdb109a3
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 1.0.1
          ioflo: Not Installed
         Jinja2: 2.9.6
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 3.5.2 (default, Nov 17 2016, 17:05:23)
   python-gnupg: Not Installed
         PyYAML: 3.12
          PyZMQ: 16.0.2
           RAET: Not Installed
          smmap: 0.9.0
        timelib: Not Installed
        Tornado: 4.5.1
            ZMQ: 4.1.6
 
System Versions:
           dist: Ubuntu 16.04 xenial
        machine: x86_64
        release: 4.9.15-x86_64-linode81
         system: Linux
        version: Ubuntu 16.04 denial
```

**Comment by ninja-** [comment]
It might be another CLI start time regression. Could you test with Python2 master and python3 minion?

**Comment by cachedout** [comment]
Based on past experience, my guess is that @ninja- is exactly right here, though we'll have to profile each piece to get to the true nature of this slowdown of course.

**Comment by DmitryKuzmenko** [comment]
I've ran both minion and master with py3, here are the difference between py2 and py3 cli:
```
(py2) $ time ./scripts/salt -l info alpha test.ping
alpha:
    True

real    0m0.427s
user    0m0.237s
sys     0m0.051s

(py3) $ time ./scripts/salt -l info alpha test.ping
alpha:
    True

real    0m0.799s
user    0m0.590s
sys     0m0.050s
```

**Comment by DmitryKuzmenko** [comment]
Looks like it's not a Salt issue. Moreover Salt works faster under Py3 but Py3 itself takes longer time to import modules. Since salt imports a lot of modules the initialization takes longer time.

To proof this I've recorded all modules imported by `salt alpha test.ping` command. Filtered out salt modules leaving the only external and built in modules. So I've got a python script consisting of the only about 800 imports. Running that file I've got the same time difference we're getting calling `salt`:
```
(py3) $ time python import.py 

real    0m0.955s
user    0m0.883s
sys     0m0.068s

(py2) $ time python ./import.py 

real    0m0.466s
user    0m0.398s
sys     0m0.067s
```

Also I've recorded the time python spends loading each of the modules to check what module has the worst regression value. Topmost three are:
```
infi - 0.2119820118 sec longer
jinja2 - 0.1406497955 sec longer
requests - 0.07497620583 sec longer
Total: ~0.43 sec longer import time.
```

The import performance tests and the full results are on my GDrive: https://goo.gl/gtnKFi
